extruct/uniform.py

```diff
@@ -1,21 +1,31 @@
 from six.moves.urllib.parse import urlparse, urljoin
 
 
-def _uopengraph(extracted):
+def _uopengraph(extracted, with_og_arr=False):
     out = []
     for obj in extracted:
-        # In order of appearance in the page
-        properties = list(reversed(obj['properties']))
-        # Ensuring that never empty value is returned if there is a duplicated
-        # property with non empty value
-        non_empty_props = {k for k, v in properties if v and v.strip()}
-        flattened = {k: v for k, v in properties
-                     if k not in non_empty_props or (v and v.strip())}
-        t = flattened.pop('og:type', None)
-        if t:
-            flattened['@type'] = t
-        flattened['@context'] = obj['namespace']
-        out.append(flattened)
+      properties = list(reversed(obj['properties']))
+      # Set of non empty properties
+      non_empty_props = {k for k, v in properties if v and v.strip()}
+      # Set of repeated properties with at least 2 non empty values
+      repeated_props = {}
+      if with_og_arr:
+        repeated_props = {k for k in non_empty_props if len([i for i,v in properties if i==k and (v and v.strip())]) > 1}
+      # Add properties that is either duplicated but has only 1 non empty value
+      # or has only empty values
+      flattened = {k: v for k, v in properties
+                       if k not in repeated_props and (k not in non_empty_props or (v and v.strip()))}
+      if with_og_arr:
```

Comment:
I think there are two options on how to make this change backwards-compatible, either add a prefix, or have a flag - I think if we have a flag like now, there is no need to add a prefix, and it's better to output a list instead of the original key here, although I'm not 100% sure which is the best API.

Reply:
Just to clarify, if we end up using the flag are you suggesting that we not use the list suffix ? Also what do you mean by "it's better to output a list instead of the original key here" ?

Reply:
> if we end up using the flag are you suggesting that we not use the list suffix? 

Right, exactly - with the flag we can change behaviour, and having result like ``"og:image": [...]`` looks more convenient to me, what do you think?

> it's better to output a list instead of the original key here

Here I meant the same, that since we're using a flag, we can use original key, for example ``"og:image"`` instead of ``"og:image_list"``.

Reply:
Ohh okay, makes sense thanks. Should I make the necessary changes and push them ?

Reply:
yes, that would be perfect, thank you!